sound/soc/soc-topology.c

```diff
@@ -2604,3 +2604,15 @@ int snd_soc_tplg_component_remove(struct snd_soc_component *comp, u32 index)
 	return !list_empty(&comp->dobj_list);
 }
 EXPORT_SYMBOL_GPL(snd_soc_tplg_component_remove);
+
+void snd_soc_tplg_route_remove(struct snd_soc_dapm_route *route)
```

Comment:
Can you make sure this is called by other drivers that use topology. Probably used in the driver remove()

Reply:
@lgirdwood 
I have to do more debug on this to confirm this.
But actually I am not clear about your question. the topology parsing only start from the entry point: snd_soc_register_card() at the machine driver register stage.
is there any other driver will do this?  the snd_soc_unregister_card()'s call-stack I will confirm whether it will call
snd_sof_free_topology() function or not.

the topology parsing's call stack is:
snd_soc_register_card()
snd_soc_instantiate_card()
soc_probe_link_components()
......
sof_pcm_probe()
snd_sof_load_topology()

Reply:
I'm asking you to check if any one else will leak memory as a result of this update and if so, they will have call route_free() when they are removed.

Reply:
I will check it.

Reply:
Hi Zhigang,
you can simple do change as below, that is enough:

/* ASoC SOF DAPM route */
struct snd_sof_route {
	struct snd_sof_dev *sdev;

-	struct snd_soc_dapm_route *route;
+	struct snd_soc_dapm_route route;
	struct list_head list;	/* list in sdev route list */

	void *private;
};

Reply:
- struct snd_soc_dapm_route *route;

   + struct snd_soc_dapm_route route;

Reply:
@keyonjie 
That is the cool solution to solve this issue.
I will rewrite this PR!
Thanks, Keyon.